util/dvsim/MsgBuckets.py

```diff
@@ -0,0 +1,218 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""
+Helper class for aggregating message categories from multiple tool runs
+"""
+import hjson
+import copy
+import logging as log
+from pathlib import Path
+
+from typing import Dict, List
+from utils import print_msg_list
+
+# Known severity levels and color map for Dvsim summary tables.
+SEV_COLMAP = {
+    'info': 'I',
+    'warning': 'W',
+    'error': 'E'
+}
+
+
+class MsgBuckets():
+    def __init__(self,
+                 bucket_dict: Dict = None,
+                 result_path: Path = None,
+                 report_sev: List[str] = ['warning', 'error'],
+                 fail_sev: List[str] = ['warning', 'error']) -> None:
+        self.buckets = {}
+        self.labels = {}
+        self.severities = {}
```

Comment:
These are very inter-related. IMO having a singular `buckets` dict that encapsulates the signatures and labels for each severity (as suggested as a comment in the previous commit) will make this simpler.

Reply:
Yeah as mentioned before I changed it due to another review comment, but given that there is more duplication now I'll change it back.